args/core/types/type_util.hpp

```diff
@@ -106,11 +86,11 @@ namespace args::core
 		hash = 0x811c9dc5;
 		uint32 prime = 0x1000193;
 		cstring name = typeName<T>();
-		for (int i = 0; i < std::strlen(name); i++)
-		{
-			byte_t value = name[i];
-			hash = hash ^ value;
+		while(*name != '\0')
```

Comment:
note that this is potentially a very dangerous approach, consider only allowing string literal hashing via `auto operator"_hash(const char *, std::size_t)` to avoid potential arbitray memory lookup. (consider what happens when someone malicious injects a non null terminated string!)
use std::string::size() on normal strings or replace it with std::string_view::size() to cover both const char* and std::string simultaneously. 
```suggestion
		for(std::string::size_type i = 0; i < name.size(); ++i) 
		//or
		for(const char n : name)
```